src/common/multi_file_reader.cpp

```diff
@@ -105,6 +242,42 @@ bool MultiFileReader::ComplexFilterPushdown(ClientContext &context, vector<strin
 		column_map.insert({get.names[get.column_ids[i]], i});
 	}
 
+	if (options.hive_file_filter) {
```

Comment:
I think this logic could be moved into its own function: In the process it would ideally become a bit clearer to the reader what we are doing here. The whole point of this feature is that instead of producing the file list up front, we perform the bind with one of the files, than expand it during the complex filter pushdown.  Effectively we are pushing down the filter into the Glob. 

Maybe create a function `vector<string> ApplyHiveGlob(const string& first_file, ....)` that performs this step?